lib/install/validate/config.go

```diff
@@ -25,106 +26,322 @@ import (
 		}
-		log.Warnf("Firewall must permit %d/tcp outbound if firewall is reenabled", rule.Port)
 	}
-	return
+
+	for _, n := range allowedNets {
+		_, aNet, err := net.ParseCIDR(n)
+		if err != nil {
+			continue
+		}
+		if aNet.Contains(checkIP.IP) {
+			return true
+		}
+	}
+	return false
+}
+
+// FirewallEnabled checks if the host firewall is enabled
+func (v *Validator) FirewallEnabled(host *object.HostSystem) (bool, error) {
+	esxfw, err := esxcli.GetFirewallInfo(host)
+	if err != nil {
+		return false, err
+	}
+	if esxfw.Enabled {
+		log.Infof("Firewall status: ENABLED on %q", host.InventoryPath)
+		return true, nil
+	}
+	log.Infof("Firewall status: DISABLED on %q", host.InventoryPath)
+	return false, nil
+}
+
+// GetMgmtIP finds the management network IP in config
+func (v *Validator) GetMgmtIP(conf *config.VirtualContainerHostConfigSpec) net.IPNet {
+	var mgmtIP net.IPNet
+	if conf != nil {
+		for _, n := range conf.ExecutorConfig.Networks {
+			if n.Network.Common.Name == "management" {
+				if n.IP != nil {
+					mgmtIP = *n.IP
+					return mgmtIP
+				}
+			}
+		}
+	}
+	return mgmtIP
+}
+
+// ManagementNetAllowed checks if the management network is allowed based
+// on the host firewall's allowed IP settings
+func (v *Validator) ManagementNetAllowed(ctx context.Context, mgmtIP net.IPNet,
+	host *object.HostSystem, requiredRule types.HostFirewallRule) (bool, error) {
+
+	fs, err := host.ConfigManager().FirewallSystem(ctx)
+	if err != nil {
+		return false, err
+	}
+	info, err := fs.Info(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	rs := object.HostFirewallRulesetList(info.Ruleset)
+	filteredRules, err := rs.EnabledByRule(requiredRule, true) // find matching rules that are enabled
+	if err != nil {                                            // rule not enabled (fw is misconfigured)
+		return false, &FirewallMisconfiguredError{Host: host.InventoryPath, Rule: requiredRule}
+	}
+	log.Debugf("filtered rules: %v", filteredRules)
+
+	// check that allowed IPs permit management IP
+	var allowedIPs []string
+	var allowedNets []string
+	for _, r := range filteredRules {
+		log.Debugf("filtered IPs: %v networks: %v allIP: %v rule: %v",
+			r.AllowedHosts.IpAddress, r.AllowedHosts.IpNetwork, r.AllowedHosts.AllIp, r.Key)
+
+		if r.AllowedHosts.AllIp { // this rule allows all hosts
+			return true, nil
+		}
+		for _, h := range r.AllowedHosts.IpAddress {
+			allowedIPs = append(allowedIPs, h)
+		}
+		for _, n := range r.AllowedHosts.IpNetwork {
+			s := fmt.Sprintf("%s/%d", n.Network, n.PrefixLength)
+			allowedNets = append(allowedNets, s)
+		}
+	}
+	if mgmtIP.IP == nil { // DHCP
+		if len(allowedIPs) > 0 || len(allowedNets) > 0 {
+			return false, &FirewallUnknownDHCPAllowedIPError{AllowedIPs: append(allowedNets, allowedIPs...),
+				Host:     host.InventoryPath,
+				Rule:     requiredRule,
+				TargetIP: mgmtIP}
+		}
+		// no allowed IPs
+		return false, &FirewallMisconfiguredError{Host: host.InventoryPath, Rule: requiredRule}
```

Comment:
If we do know the subnet at this point, then yes we should check for it here. In many case though, we don't know, and it is not possible to check here. We may just put out a warning here instead of an error.